habitat_baselines/il/data/data.py

```diff
@@ -0,0 +1,544 @@
+import copy
+import os
+from typing import List, Tuple
+
+import cv2
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+from habitat import logger
+from habitat.core.simulator import ShortestPathPoint
+from habitat_baselines.il.models.models import MultitaskCNN
+
+
+class EQADataset(Dataset):
+    """Pytorch dataset for Embodied Q&A (both VQA and NAV)"""
+
+    def __init__(
+        self, env, config, input_type, num_frames=5, max_controller_actions=5,
+    ):
+        """
+        Args:
+            env (habitat.Env): Habitat environment
+            config: Config
+            input_type (string): Type of model being trained ("vqa", "pacman")
+            num_frames (int): number of frames used as input to VQA model
+            max_controller_actions (int):
+        """
+        self.env = env
+        self.config = config.TASK_CONFIG
+        self.input_type = input_type
+        self.num_frames = num_frames
+
+        self.sim = self.env.sim
+        self.episodes = self.env._dataset.episodes
+
+        self.q_vocab = self.env._dataset.question_vocab
+        self.ans_vocab = self.env._dataset.answer_vocab
+
+        self.eval_save_results = config.EVAL_SAVE_RESULTS
+
+        if self.config.DATASET.SPLIT == config.EVAL.SPLIT:
+            self.mode = "val"
+        else:
+            self.mode = "train"
+
+        self.frame_dataset_path = config.FRAME_DATASET_PATH
+        self.only_vqa_task = config.ONLY_VQA_TASK
+        self.disk_cache_exists = False
+        # first scene that is loaded by default
+        self.temp_scene_id = self.episodes[0].scene_id
+
+        # [TODO] can be done in mp3d_eqa_dataset while loading dataset
+        self.calc_max_length()
+        self.restructure_ans_vocab()
+
+        self.scene_ids = []
+        self.scene_episode_dict = {}
+
+        # preparing a dict that stores list of episodes for each scene
+        for ep in tqdm(self.episodes):
+            if ep.scene_id not in self.scene_ids:
+                self.scene_ids.append(ep.scene_id)
+                self.scene_episode_dict[ep.scene_id] = [ep.episode_id]
+            else:
+                self.scene_episode_dict[ep.scene_id].append(ep.episode_id)
+
+        if self.input_type == "pacman":
+
+            cnn_kwargs = {
+                "num_classes": 40,
+                "pretrained": True,
+                "checkpoint_path": config.EDFE_CKPT_PATH,
+            }
+            self.cnn = MultitaskCNN(**cnn_kwargs)
+            self.cnn.eval()
+            self.cnn.cuda()
```

Comment:
Use `to(device=self.device)` (or similar) instead of `cuda()`.  That way it will work on just CPU also.